src/SonarQube.Client.Tests/Requests/Api/V7_20/GetIssuesRequestWrapperTests.cs

```diff
@@ -74,44 +83,76 @@ public async Task InvokeAsync_HasIssueKeys_ExpectedPropertiesArePassedInASingleR
-            actualQueryString.Contains("&rules=rule1").Should().BeTrue();
-            actualQueryString.Contains("&components=project1").Should().BeTrue();
-            actualQueryString.Contains("types").Should().BeFalse();
+            CheckExpectedQueryStringsParameters(componentPropertyName, handlerMock, 0, "aaaProject", "xStatus", "yBranch", "rule1", "project1", expectedKeys: issueKeys);
         }
 
-        private static GetIssuesRequestWrapper CreateTestSubject(string projectKey, string statusesToRequest, string branch, string[] issueKeys, string ruleId, string componentKey)
+        private static IGetIssuesRequest CreateTestSubject(string componentPropertyName, string projectKey, string statusesToRequest, string branch, string[] issueKeys, string ruleId, string componentKey)
         {
-            var testSubject = new GetIssuesRequestWrapper
+            return componentPropertyName switch
             {
-                Logger = new TestLogger(),
-                ProjectKey = projectKey,
-                Statuses = statusesToRequest,
-                Branch = branch,
-                IssueKeys = issueKeys,
-                RuleId = ruleId,
-                ComponentKey = componentKey
+                SonarQubeComponentPropertyName => new GetIssuesRequestWrapper<GetIssuesWithComponentSonarQubeRequest>
+                {
+                    Logger = new TestLogger(),
+                    ProjectKey = projectKey,
+                    Statuses = statusesToRequest,
+                    Branch = branch,
+                    IssueKeys = issueKeys,
+                    RuleId = ruleId,
+                    ComponentKey = componentKey
+                },
+                SonarCloudComponentPropertyName => new GetIssuesRequestWrapper<GetIssuesWithComponentSonarCloudRequest>
+                {
+                    Logger = new TestLogger(),
+                    ProjectKey = projectKey,
+                    Statuses = statusesToRequest,
+                    Branch = branch,
+                    IssueKeys = issueKeys,
+                    RuleId = ruleId,
+                    ComponentKey = componentKey
+                },
+                _ => throw new ArgumentOutOfRangeException()
             };
-
-            return testSubject;
         }
 
-        private static void CheckExpectedQueryStringsParameters(Mock<HttpMessageHandler> handlerMock, int invocationIndex,
-            string expectedProject, string expectedStatues, string expectedBranch, string expectedTypes, string expectedRule, string expectedComponent)
+        private static void CheckExpectedQueryStringsParameters(string componentKeyName,
+            Mock<HttpMessageHandler> handlerMock,
+            int invocationIndex,
+            string expectedProject,
+            string expectedStatues,
+            string expectedBranch,
+            string expectedRule,
+            string expectedComponent,
```

Comment:
For sake of simplicity, do you need to bother with these parameters? I think values for most of them can be just hardcoded in the assertion. As the test is small, and individual UTs can be aligned on "&projects=project1"